R/ys-tidy.R

```diff
@@ -265,20 +276,80 @@ ys_join <- function(left, right, ...) {
 #' 
 #' tail(ans)
 #' 
-#' @return A `yspec` object with renamed columns
+#' data <- ys_help$data()
+#' 
+#' ys_rename(data, spec, AGE:HT)
+#' 
+#' @return 
+#' A `yspec` object or `data.frame` with renamed columns, depending on what was
+#' passed as `.x`.
 #' 
 #' @seealso [ys_join()], [ys_filter()], [ys_select()]
 #' @md
 #' @export
-ys_rename <- function(.x, ...) {
-  assert_that(is_yspec(.x))
+ys_rename <- function(.x, ... ) UseMethod("ys_rename")
+
+#' @rdname ys_rename
+#' @export
+ys_rename.yspec<- function(.x, ...) {
   re <- eval_rename(expr(c(...)), as.list(.x))
   names(.x)[re] <- names(re)
   for(i in seq_along(re)) {
     .x[[re[[i]]]][["col"]] <- names(re)[[i]]
   }
   .x
 }
+#' @rdname ys_rename
+#' @export
+ys_rename.data.frame <- function(.x, .spec, ..., .title_case = TRUE, .unit = FALSE) {
+  assert_that(is_yspec(.spec))
+  re <- eval_select(expr(c(...)), .x)
+  if(isTRUE(.unit)) {
+    short <- ys_get_short_unit(.spec, title_case = .title_case)  
```

Comment:
Can users still pass in the `short_max` argument for these two ys_get_short functions?

Reply:
I can pass that through too.

Reply:
I think that would be helpful both here and in the ys_recode section (or can you pass it through the ... section in ys_recode?)